.github/actions/torchserve-setup/action.yml

```diff
@@ -0,0 +1,306 @@
+      shell: bash
+      run: |
+        # Wait for TorchServe
+        echo "Waiting for TorchServe to be ready..."
+        MAX_ATTEMPTS=30
+        for i in $(seq 1 $MAX_ATTEMPTS); do
+          if curl -s http://localhost:${{ inputs.inference-port }}/ping | grep -q "Healthy"; then
+            echo "✓ TorchServe is ready!"
+            break
+          fi
+          echo "Waiting for TorchServe... ($i/$MAX_ATTEMPTS)"
+          [ $i -eq $MAX_ATTEMPTS ] && { echo "ERROR: TorchServe not responding"; exit 1; }
+          sleep 5
+        done
+
+        # Find TorchServe container
+        CONTAINER_ID=$(docker ps --format "{{.ID}}" --filter "publish=${{ inputs.inference-port }}" | head -1)
+        [ -z "$CONTAINER_ID" ] && { echo "ERROR: TorchServe container not found"; docker ps; exit 1; }
+        echo "Found TorchServe container: $CONTAINER_ID"
+
+        # Install dependencies in container
+        echo "Installing dependencies in container (this may take a moment)..."
+        docker exec $CONTAINER_ID pip install -q --no-cache-dir \
+          transformers==4.35.0 \
+          datasets \
+          nltk \
+          safetensors
+
+        # Download NLTK data
+        docker exec $CONTAINER_ID python -c "
+        import nltk
+        try:
+            nltk.download('punkt', quiet=True)
+            nltk.download('punkt_tab', quiet=True)
+        except:
+            pass
+        " 2>/dev/null
+        echo "✓ Dependencies installed"
+
+    - name: Check Model Status
+      id: check-model
+      if: inputs.action == 'setup-model' || inputs.action == 'full-setup'
+      shell: bash
+      run: |
+        # Check if model is already loaded
+        MGMT_URL="http://localhost:${{ inputs.management-port }}"
+        MODEL_STATUS=$(curl -s "${MGMT_URL}/models" 2>/dev/null || echo "[]")
+
+        if echo "$MODEL_STATUS" | grep -q "${{ inputs.model-name }}"; then
+          echo "Model '${{ inputs.model-name }}' is already loaded"
+          echo "model_loaded=true" >> $GITHUB_OUTPUT
+        else
+          echo "Model '${{ inputs.model-name }}' not loaded"
+          echo "model_loaded=false" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Deploy Model
+      if: (inputs.action == 'setup-model' || inputs.action == 'full-setup') && steps.check-model.outputs.model_loaded == 'false'
+      shell: bash
+      run: |
+        set -e
+
+        # Create working directories
+        mkdir -p model_store model_files
+
+        # Prepare handler
+        if [ -f "${{ inputs.handler-path }}" ]; then
+          echo "Using handler from: ${{ inputs.handler-path }}"
+          cp "${{ inputs.handler-path }}" handler.py
+        else
+          echo "ERROR: Handler file not found at ${{ inputs.handler-path }}"
+          exit 1
+        fi
+
+        # Download model files from HuggingFace
+        echo "Downloading model files..."
+        BASE_URL="https://huggingface.co/opensearch-project/opensearch-semantic-highlighter-v1/resolve/main"
```

Comment:
Should be passed as env variable or some dynamic way. That way we can deploy any model

Reply:
Sure